zebra-chain/src/parameters/network/subsidy.rs

```diff
@@ -503,10 +545,54 @@ pub fn funding_stream_address_period<N: ParameterSubsidy>(height: Height, networ
 
     let height_after_first_halving = height - network.height_for_first_halving();
 
-    let address_period = (height_after_first_halving + POST_BLOSSOM_HALVING_INTERVAL)
-        / FUNDING_STREAM_ADDRESS_CHANGE_INTERVAL;
+    let address_period = (height_after_first_halving + network.post_blossom_halving_interval())
+        / network.funding_stream_address_change_interval();
 
     address_period
         .try_into()
         .expect("all values are positive and smaller than the input height")
 }
+
+/// The first block height of the halving at the provided halving index for a network.
+///
+/// See `Halving(height)`, as described in [protocol specification §7.8][7.8]
+///
+/// [7.8]: https://zips.z.cash/protocol/protocol.pdf#subsidies
+pub fn height_for_halving_index(halving_index: u32, network: &Network) -> Option<Height> {
```

Comment:
Could we remove `index` from the function/argument names? It seems needlessly confusing (I think I may have named it, but I made a mistake).

```suggestion
pub fn height_for_halving(halving: u32, network: &Network) -> Option<Height> {
```